primitives/state-machine/src/ext.rs

```diff
@@ -72,38 +91,83 @@ impl<B: error::Error, E: error::Error> error::Error for Error<B, E> {
 }
 
 /// Wraps a read-only backend, call executor, and current overlayed changes.
+#[cfg(feature = "std")]
 pub struct Ext<'a, H, N, B>
 	where
 		H: Hasher,
 		B: 'a + Backend<H>,
 		N: crate::changes_trie::BlockNumber,
 {
-	/// The overlayed changes to write to.
-	overlay: &'a mut OverlayedChanges,
+	/// Inner Ext (change overlay and backend).
+	inner: ExtInner<'a, H, B, N>,
 	/// The overlayed changes destined for the Offchain DB.
 	offchain_overlay: &'a mut OffchainOverlayedChanges,
-	/// The storage backend to read from.
-	backend: &'a B,
-	/// The cache for the storage transactions.
-	storage_transaction_cache: &'a mut StorageTransactionCache<B::Transaction, H, N>,
 	/// Changes trie state to read from.
 	changes_trie_state: Option<ChangesTrieState<'a, H, N>>,
-	/// Pseudo-unique id used for tracing.
-	pub id: u16,
 	/// Dummy usage of N arg.
 	_phantom: std::marker::PhantomData<N>,
 	/// Extensions registered with this instance.
 	extensions: Option<&'a mut Extensions>,
 }
 
+/// Basis implementation for `Externalities` trait.
+pub struct ExtInner<'a, H, B, N>
```

Comment:
Why do we need this split?

Reply:
I initially wanted to avoid putting some `#[cfg(feature(std))]` all over the code, but recent change put some all over so I guess at this point I can just remove the field from Ext and have different implementation for a few function.